### PR TITLE
refactor(button): updates button colors to match Figma

### PR DESCRIPTION
**Pull request by caripizza**
**Related Issue:** #2456, #2474, #2472, #2744

## Summary
Updates calcite-button's colors per the Figma > _Infastructures > "Button Colors" design. (Dependents calcite-fab and calcite-split-button display the new colors as well.)

Additional updates (follow-up tweaks now that pr #2533 is in):
- ensures button loader and icons are 16px in all scales
- ensures a large calcite-split-button's dropdown-trigger icon is only 16px
- ensures margins between text/icons in buttons are 8/12/16px per Figma
- ignores invalid appearance values in FAB
- ensures `--calcite-ui-inverse` global vars have unique values for light/dark themes

### CODE REVIEW

**Comment by caripizza** [comment]
@bstifle can you take a look at this one to be sure I'm on the right track for the new colors? Thank you!

**Comment by bstifle** [comment]
sure thing!!

**Comment by bstifle** [comment]
@caripizza NICE WORK!

Only thing, before 1.0 (idk if we need to do this in this PR or not) blue should be `brand` and red should be `danger`

**Comment by caripizza** [comment]
> Only thing, before 1.0 (idk if we need to do this in this PR or not) blue should be brand and red should be danger

I'll create a discussion issue for this now so we can hash that out (different pr for sure).

**Comment by caripizza** [comment]
This one's still waiting on a code review

File: src/components/calcite-button/calcite-button.tsx [hunk 1]
```diff
@@ -155,7 +157,7 @@ export class CalciteButton {
         class={{ [CSS.icon]: true, [CSS.iconStart]: true }}
         flipRtl={this.iconFlipRtl === "start" || this.iconFlipRtl === "both"}
         icon={this.iconStart}
-        scale="s"
+        scale={iconScale}
```
Comment (julio8a) [inline src/components/calcite-button/calcite-button.tsx:160]: From a conversation we had with Bryan, I though that S, M, or L button icons need to be 16px. This is what's on Figma too. 

CC'ing @bstifle

**Review by julio8a — COMMENTED** [review]
From the little code I know, everything else looks good.

**Comment by caripizza** [comment]
Are we good to go on this one?

**Review by julio8a — APPROVED** [review]
LGTM!

**Pull request merged**